filecoin-proofs/src/api/seal.rs

```diff
@@ -1130,3 +1135,118 @@ pub fn verify_batch_seal<Tree: 'static + MerkleTreeTrait>(
     info!("verify_batch_seal:finish");
     result
 }
+
+/// Generate the merkle tree on top of the replica (TreeRLast).
+///
+/// The generated trees are stored in `output_dir`, usually the cache directory. The `replica_path`
+/// point to the replica where the tree should be built upon. The `sector_size` is in bytes.
+pub fn generate_tree_r_last<O, R, TreeR: 'static + MerkleTreeTrait>(
+    sector_size: u64,
+    replica_path: R,
+    output_dir: O,
+) -> Result<<TreeR::Hasher as Hasher>::Domain>
+where
+    O: AsRef<Path>,
+    R: AsRef<Path>,
+{
+    let leaf_count = sector_size as usize / NODE_SIZE;
+    let tree_count = get_base_tree_count::<TreeR>();
+    let nodes_count = leaf_count / tree_count;
```

Comment:
It may be a good idea to name this `base_tree_leafs` or something similar (I know that the name `nodes_count` is used in `StackedDrg::generate_tree_...`, but that naming is bad imo).

Isn't this value the same as `base_tree_leafs` below?

Reply:
This comment also applies to the below function.

Reply:
> Isn't this value the same as `base_tree_leafs` below?

No it's the binary vs. the octa tree.

Reply:
Oh, right. Thanks